ci/fetch_binaries.py

```diff
@@ -0,0 +1,143 @@
+        )
+        .decode()
+        .splitlines()
+    )
+    tags = [tag for tag in tags if tag.startswith(f"{tag_prefix}-")]
+
+    latest_tag = None
+    latest_date = None
+
+    for tag in tags:
+        date_str = extract_date(tag, tag_prefix)
+        date = datetime.strptime(
+            date_str, "%y%m%d" if len(date_str) == 6 else "%y%m%d%H%M"
+        )
+        if latest_date is None or date > latest_date:
+            latest_tag = tag
+            latest_date = date
+
+    if latest_tag:
+        message = (
+            subprocess.check_output(
+                ["git", "-C", PROJECT_ROOT, "tag", "-l", "-n1", latest_tag]
+            )
+            .decode()
+            .strip()
+            .split(" ", 1)[1]
+        )
+        return latest_tag, message
+    else:
+        return None, None
+
+
+def get_remote_path() -> str:
+    LIBTELIO_BUILD_PROJECT_ID = 6299
+    libtelio_env_sec_gitlab_repository = os.environ.get(
+        "LIBTELIO_ENV_SEC_GITLAB_REPOSITORY", None
+    )
+
+    if libtelio_env_sec_gitlab_repository is None:
+        raise ValueError("LIBTELIO_ENV_SEC_GITLAB_REPOSITORY not set.")
+
+    return f"https://{libtelio_env_sec_gitlab_repository}/api/v4/projects/{LIBTELIO_BUILD_PROJECT_ID}"
+
+
+def get_api(path, timeout=300):
+    with requests.get(
+        get_remote_path() + path,
+        headers={"PRIVATE-TOKEN": TOKEN if TOKEN else ""},
+        timeout=timeout,
+    ) as request:
+        request.raise_for_status()
+        response_string = request.content.decode("utf-8")
+        return response_string
+
+
+def get_artifacts(path_to_save, job, timeout=300, unzip=False):
+    full_path = path_to_save + job["artifacts_file"]["filename"]
+
+    print("Getting artficats for ", job["name"], ", filename: ", full_path)
+
+    r = requests.get(
+        get_remote_path() + "/jobs/" + str(job["id"]) + "/artifacts",
+        headers={"PRIVATE-TOKEN": TOKEN if TOKEN else ""},
+        timeout=timeout,
+    )
+    with open(str(full_path), "wb") as f:
+        f.write(r.content)
+
+    with zipfile.ZipFile(full_path, "r") as zip_ref:
+        zip_ref.extractall(path_to_save)
+
+
+def get_pipeline_build_artifacts(pipeline_id, path_to_save, target_arch, target_os):
+    for job in json.loads(
+        get_api(
+            (
+                f"/pipelines/{pipeline_id}/jobs?per_page=100&include_retried=true&scope=success"
+            )
+        )
+    ):
+        if job["stage"] == "build":
+            if target_os == "uniffi" and job["name"] == "uniffi-bindings":
+                get_artifacts(path_to_save, job)
+                return True
+            else:
+                if (
+                    target_os in job["name"] if target_os is not None else True
+                ) and target_arch in job["name"]:
+                    get_artifacts(path_to_save, job, unzip=True)
+                    return True
+
+    return False
+
+
+def fetch_build_artifacts(tag_prefix, target_os, target_arch, token, download_dir):
```

Comment:
I think would be cleaner if we had some sort of config object where these parameters are stored and then pass it to `get_pipeline_build_artifacts(config)` where the data processing is done, globals would be avoided. Also wondering if we would be better to call `get_pipeline_build_artifacts()` directly from `build_libtelio.py`, why whould we need to have both `fetch_build_artifacts()` and `get_pipeline_build_artifacts()`, I guess this is more of a design question than a change request..

Reply:
This naming is a leftover, but yeah, I agree that it is confusing (the _api_ of that file should only include `fetch_build_artifacts()`).  I'll change it to an object based fetch.

Reply:
Fixed